src/test/scala/org/neo4j/spark/DataSourceReaderTSE.scala

```diff
@@ -1296,6 +1298,213 @@ class DataSourceReaderTSE extends SparkConnectorScalaBaseTSE {
     assertEquals(10, df.count())
   }
 
+  @Test
+  def testShouldReturnJustTheSelectedFieldWithNode(): Unit = {
+    val total = 100
+    val fixtureQuery: String =
+      s"""UNWIND range(1, $total) as id
+         |CREATE (pr:Product {id: id, name: 'Product ' + id})
+         |RETURN *
+    """.stripMargin
+
+    SparkConnectorScalaSuiteIT.session()
+      .writeTransaction(
+        new TransactionWork[ResultSummary] {
+          override def execute(tx: Transaction): ResultSummary = tx.run(fixtureQuery).consume()
+        })
+
+    val df = ss.read
+      .format(classOf[DataSource].getName)
+      .option("url", SparkConnectorScalaSuiteIT.server.getBoltUrl)
+      .option("labels", "Product")
+      .load
+      .select("name")
+
+    df.count()
+
+    assertEquals(Set("name"), df.columns.toSet)
+  }
+
+  @Test
+  def testShouldReturnJustTheSelectedFieldWithNodeAndWeirdColumnName(): Unit = {
+    val total = 100
+    val fixtureQuery: String =
+      s"""UNWIND range(1, $total) as id
+         |CREATE (pr:Product {id: id, `(╯°□°)╯︵ ┻━┻`: 'Product ' + id})
+         |RETURN *
+    """.stripMargin
+
+    SparkConnectorScalaSuiteIT.session()
+      .writeTransaction(
+        new TransactionWork[ResultSummary] {
+          override def execute(tx: Transaction): ResultSummary = tx.run(fixtureQuery).consume()
+        })
+
+    val df = ss.read
+      .format(classOf[DataSource].getName)
+      .option("url", SparkConnectorScalaSuiteIT.server.getBoltUrl)
+      .option("labels", "Product")
+      .load
+      .select("`(╯°□°)╯︵ ┻━┻`")
+
+    df.count()
+
+    assertEquals(Set("(╯°□°)╯︵ ┻━┻"), df.columns.toSet)
+  }
+
+  @Test
+  def testShouldReturnJustTheSelectedFieldWithRelationship(): Unit = {
+    val total = 100
+    val fixtureQuery: String =
+      s"""UNWIND range(1, $total) as id
+         |CREATE (pr:Product {id: id * rand(), name: 'Product ' + id})
+         |CREATE (pe:Person {id: id, fullName: 'Person ' + id})
+         |CREATE (pe)-[:BOUGHT{when: rand(), quantity: rand() * 1000}]->(pr)
+         |RETURN *
+    """.stripMargin
+
+    SparkConnectorScalaSuiteIT.session()
+      .writeTransaction(
+        new TransactionWork[ResultSummary] {
+          override def execute(tx: Transaction): ResultSummary = tx.run(fixtureQuery).consume()
+        })
+
+    val df = ss.read
+      .format(classOf[DataSource].getName)
+      .option("url", SparkConnectorScalaSuiteIT.server.getBoltUrl)
+      .option("relationship", "BOUGHT")
+      .option("relationship.source.labels", "Product")
+      .option("relationship.target.labels", "Person")
+      .load
+      .select("`source.name`", "`<source.id>`")
+
+    df.count()
+
+    assertEquals(Set("source.name", "<source.id>"), df.columns.toSet)
```

Comment:
`Seq` instead of `Set`? we want also to check the order here

Reply:
same as in the other tests

Reply:
maybe here you can leave the tests as they already are and add similar in 4.x?

Reply:
problem is that in 3.5 order is not guaranteed, I will do the same test in 4.x